src/app/InteractionModelEngine.h

```diff
@@ -133,19 +155,21 @@ class InteractionModelEngine : public Messaging::ExchangeDelegate
 
     InteractionModelEngine(void);
 
-    CHIP_ERROR Init(Messaging::ExchangeManager * apExchangeMgr);
+    CHIP_ERROR Init(Messaging::ExchangeManager * apExchangeMgr, void * const apAppState, const EventCallback aEventCallback);
 
     void Shutdown();
 
-    CHIP_ERROR DeregisterClusterCommandHandler(chip::ClusterId aClusterId, chip::CommandId aCommandId,
-                                               Command::CommandRoleId aCommandRoleId);
-    CHIP_ERROR RegisterClusterCommandHandler(chip::ClusterId aClusterId, chip::CommandId aCommandId,
-                                             Command::CommandRoleId aCommandRoleId, CommandCbFunct aDispatcher);
-
     Messaging::ExchangeManager * GetExchangeManager(void) const { return mpExchangeMgr; };
 
     CHIP_ERROR NewCommandSender(CommandSender ** const apComandSender);
 
+    CHIP_ERROR NewReadClient(ReadClient ** const apReadClient, chip::app::ReadClient::EventCallback aEventCallback,
```

Comment:
Same comment as above here: callbach + void* is better represented by a delegate since that is typesafe.

Reply:
Done, thanks